x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -168,38 +165,42 @@ public static Query doTranslate(ScalarFunction f, TranslatorHandler handler) {
         }
     }
 
-    public static class CaseSensitiveScalarFunctions extends ExpressionTranslator<CaseSensitiveScalarFunction> {
+    public static class CaseInsensitiveScalarFunctions extends ExpressionTranslator<CaseInsensitiveScalarFunction> {
 
         @Override
-        protected Query asQuery(CaseSensitiveScalarFunction f, TranslatorHandler handler) {
-            return f.isCaseSensitive() ? doTranslate(f, handler) : null;
+        protected Query asQuery(CaseInsensitiveScalarFunction f, TranslatorHandler handler) {
+            return doTranslate(f, handler);
         }
 
-        public static Query doTranslate(CaseSensitiveScalarFunction f, TranslatorHandler handler) {
-            Expression field = null;
-            Expression constant = null;
-
-            if (f instanceof StringContains) {
-                StringContains sc = (StringContains) f;
-                field = sc.string();
-                constant = sc.substring();
-            } else if (f instanceof EndsWith) {
-                EndsWith ew = (EndsWith) f;
-                field = ew.input();
-                constant = ew.pattern();
-            } else {
-                return null;
+        public static Query doTranslate(CaseInsensitiveScalarFunction f, TranslatorHandler handler) {
+            Query q = ExpressionTranslators.Scalars.doKnownTranslate(f, handler);
+            if (q != null) {
+                return q;
             }
 
-            if (field instanceof FieldAttribute && constant.foldable()) {
-                String targetFieldName = handler.nameOf(((FieldAttribute) field).exactAttribute());
-                String substring = (String) constant.fold();
-                String query = "*" + substring + (f instanceof StringContains ? "*" : "");
+            if (f instanceof BinaryComparisonCaseInsensitiveFunction) {
+                BinaryComparisonCaseInsensitiveFunction bccif = (BinaryComparisonCaseInsensitiveFunction) f;
 
-                return new WildcardQuery(f.source(), targetFieldName, query);
-            }
+                String targetFieldName = null;
+                String wildcardQuery = null;
 
-            return null;
+                Expression field = bccif.left();
+                Expression constant = bccif.right();
+
+                if (field instanceof FieldAttribute && constant.foldable()) {
+                    targetFieldName = handler.nameOf(((FieldAttribute) field).exactAttribute());
+                    String string = (String) constant.fold();
+
+                    if (f instanceof StringContains) {
+                        wildcardQuery = "*" + string + "*";
+                    } else if (f instanceof EndsWith) {
+                        wildcardQuery = "*" + string;
```

Comment:
The String is already internalized by the JVM however I've extracted inside the utils class since we use it there.
The issue with extracting constants is that if they aren't used consistently, they just occupy space.